pkg/network/tracer/connection/ebpfless/payload.go

```diff
@@ -0,0 +1,113 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+//go:build linux
+
+// Package ebpfless contains supporting code for the ebpfless tracer
+package ebpfless
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/google/gopacket"
+	"github.com/google/gopacket/layers"
+)
+
+var errZeroLengthUDPPacket = errors.New("UDP packet with length 0")
+var errZeroLengthIPPacket = errors.New("IP packet with length 0")
+
+// Layers holds a set of network layers for a packet
+type Layers struct {
+	IP4 *layers.IPv4
+	IP6 *layers.IPv6
+	UDP *layers.UDP
+	TCP *layers.TCP
+}
+
+// Reset resets a Layers object
+func (l *Layers) Reset(ip4 *layers.IPv4, ip6 *layers.IPv6, udp *layers.UDP, tcp *layers.TCP) {
+	l.IP4 = ip4
+	l.IP6 = ip6
+	l.UDP = udp
+	l.TCP = tcp
+}
+
+// PayloadLen returns the length of the application
+// payload given the set of layers in `Layers`
+func (l Layers) PayloadLen() (uint16, error) {
+	if l.UDP != nil {
```

Comment:
this gets called in the tcp processing path as well.. I've noticed locally it triggers because the UDP layer is not nil but it is empty, which bails out of the TCP connection parsing. perhaps we should do 
```
if len(l.UDP.Payload) > 0 || len(l.UDP.Contents) > 0
```

Reply:
I removed `epbfless.Layers`, so this should happen anymore.